packages/Core/iterable.candy

```diff
@@ -128,22 +132,37 @@ any iterable tester :=
 
 ## Filtering
 
-# whereWithStateHelper = { whereWithStateHelper iterable state tester ->
-#   iterable | result.map { it ->
-#     [currentItem, remainingIterable] = it
-#     (
-#       mapper currentItem,
-#       mapHelper mapHelper remainingIterable mapper,
-#     )
-#   }
-# }
-# whereWithState iterable initialState tester :=
-#   needs (is iterable)
-#   needs (function.is2 tester)
-#   whereWithStateHelper whereWithStateHelper iterable initialState tester
-# where iterable tester :=
-#   needs (is iterable)
-#   needs (function.is1 tester)
+whereHelper = { whereHelper iterable tester ->
+  newWithState iterable { iterable ->
```

Comment:
Does this work? I know I named the function like this initially, but at the top of this file, it's now called `generateWithState`.
Same applies below